gattc_linux.go

```diff
@@ -257,12 +261,22 @@ func (c DeviceCharacteristic) EnableNotifications(callback func(buf []byte)) err
 			for sig := range c.property {
 				if sig.Name == "org.freedesktop.DBus.Properties.PropertiesChanged" {
 					interfaceName := sig.Body[0].(string)
-					if interfaceName != "org.bluez.GattCharacteristic1" {
+
```

Comment:
Maybe change this entire bit of code to:

```go
switch {
case interfaceName == "org.bluez.Device1" && sig.Path == devicePath:
	changes := sig.Body[1].(map[string]dbus.Variant)

	if connected, ok := changes["Connected"].Value().(bool); ok && !connected {
		c.EnableNotifications(nil)
		return
	}
case interfaceName != "org.bluez.GattCharacteristic1":
	continue
case sig.Path != c.characteristic.Path():
	continue
}
```

Reply:
Yeah, you're right. A switch statement is better there. I just changed it :)